test/hw2/test_task4.py

```diff
@@ -0,0 +1,14 @@
+from All_home_works.hw2.task4 import cache
+
+
+def test_cache():
+    some = 100, 200
+
+    cache_fun = cache(func)
+    val1 = cache_fun(*some)
+    val2 = cache_fun(*some)
+    assert val1 == val2
```

Comment:
You called the same function twice and got the same result.  I'm not sure if it is cached.

Reply:
Each time you change something ask yourself, will test failed if you replace your function with

```python
def cache(fun):
    return fun
```

If test is not failed after that it does not test caching.

Reply:
Oh. If I understood clearly, this test will fix it:
assert val1 is val2

Reply:
Oh. It was in descriptions, sorry. But of course, it is good, because I came to it myself.